examples/topology_extraction_example.py

```diff
@@ -0,0 +1,33 @@
+import stk
+import stko
+
+
+def main():
+    bb1 = stk.BuildingBlock('NCCN', [stk.PrimaryAminoFactory()])
+    bb2 = stk.BuildingBlock(
+        smiles='O=CC(C=O)C=O',
+        functional_groups=[stk.AldehydeFactory()],
+    )
+    cage1 = stk.ConstructedMolecule(
+        topology_graph=stk.cage.FourPlusSix((bb1, bb2)),
+    )
+
+    disconnections = []
+    for bi in cage1.get_bond_infos():
+        if bi.get_building_block() is None:
+            a1id = bi.get_bond().get_atom1().get_id()
+            a2id = bi.get_bond().get_atom2().get_id()
+            disconnections.append(sorted((a1id, a2id)))
```

Comment:
I would use `disconnections.extend(sorted(a1id, a2id))`. It's not clear to me why you're sorting here though.
Using extend here means that you do not have any issues with making your `ids_to_disconnect_flat` in `TopologyExtractor.extract_topology()`

Also, I think the flatting, even if it was necessary, should have happened here, not inside `TopologyExtractor.extract_topology()`, because having the `extract_topology()` method require an iterable of tuples for a parameter called `atom_ids` is confusing.

Reply:
I see this. but I need both the resultof `extend` and `append` for the code to work. So I sorted this with two arguments.

Thanks.